src/foremast/runner.py

```diff
@@ -268,29 +278,60 @@ def cleanup(self):
 def prepare_infrastructure():
     """Entry point for preparing the infrastructure in a specific env."""
     runner = ForemastRunner()
-
     runner.write_configs()
-    runner.create_app()
 
+    try:
+        pipeline_type = runner.configs['pipeline']['type']
+        if pipeline_type is None or pipeline_type.isspace():
+            raise ForemastError("pipeline.type is required")
+    except KeyError:
+        raise ForemastError("pipeline.type is required")
+
+    if pipeline_type in consts.GCP_TYPES:
```

Comment:
Maybe add backward compat logic.

If GCP_TYPES is null and AWS_TYPES is null flip to old config logic.

This saves our users who accidentally do major upgrades automatically from breaking changes :)

Reply:
If the older config options `types` and `manual_types` are set in the Foremast config, a deprecation warning is logged and those values are mapped to the new `AWS_*` equivalents. Commit: [b3aa70](https://github.com/foremast/foremast/pull/413/commits/b3aa70fbd097334ab29f4e490357c6c6f7b7007a)